tests/test_spatial.py

```diff
@@ -0,0 +1,224 @@
+pytest autouse fixture.
+"""
+
+import contextlib
+import io
+from pathlib import Path
+
+import f90nml
+import pytest
+import yaml
+
+from benchcab import internal
+from benchcab.model import Model
+from benchcab.spatial import SpatialTask, get_spatial_tasks
+from benchcab.utils.repo import Repo
+
+
+@pytest.fixture()
+def mock_repo():
+    class MockRepo(Repo):
+        def __init__(self) -> None:
+            self.branch = "test-branch"
+            self.revision = "1234"
+
+        def checkout(self, verbose=False):
+            pass
+
+        def get_branch_name(self) -> str:
+            return self.branch
+
+        def get_revision(self) -> str:
+            return self.revision
+
+    return MockRepo()
+
+
+@pytest.fixture()
+def model(mock_subprocess_handler, mock_repo):
+    """Returns a `Model` instance."""
+    _model = Model(
+        model_id=1,
+        repo=mock_repo,
+        patch={"cable": {"some_branch_specific_setting": True}},
+    )
+    _model.subprocess_handler = mock_subprocess_handler
+    return _model
+
+
+@pytest.fixture()
+def task(model, mock_subprocess_handler):
+    """Returns a mock `SpatialTask` instance."""
+    _task = SpatialTask(
+        model=model,
+        met_forcing_name="crujra_access",
+        met_forcing_payu_experiment="https://github.com/CABLE-LSM/cable_example.git",
+        sci_conf_id=0,
+        sci_config={"cable": {"some_setting": True}},
+    )
+    _task.subprocess_handler = mock_subprocess_handler
+    return _task
+
+
+class TestGetTaskName:
+    """Tests for `SpatialTask.get_task_name()`."""
+
+    def test_task_name_convention(self, task):
+        """Success case: check task name convention."""
+        assert task.get_task_name() == "crujra_access_R1_S0"
+
+
+class TestConfigureExperiment:
+    """Tests for `SpatialTask.configure_experiment()`."""
+
+    @pytest.fixture(autouse=True)
+    def _create_task_dir(self):
+        task_dir = internal.SPATIAL_TASKS_DIR / "crujra_access_R1_S0"
+        task_dir.mkdir(parents=True)
+        (task_dir / "config.yaml").touch()
+        (task_dir / "cable.nml").touch()
+
+    def test_payu_config_parameters(self, task):
+        """Success case: check config.yaml parameters."""
+        task.configure_experiment(payu_config={"some_parameter": "foo"})
+        config_path = internal.SPATIAL_TASKS_DIR / task.get_task_name() / "config.yaml"
+        with config_path.open("r", encoding="utf-8") as file:
+            config = yaml.safe_load(file)
+        assert config["exe"] == str(
+            (
+                internal.SRC_DIR / "test-branch" / "offline" / internal.CABLE_MPI_EXE
+            ).absolute()
+        )
+        assert config["input"] == [
+            str(
+                internal.CABLE_AUX_DIR
+                / Path("core/biogeophys/def_veg_params_zr_clitt_albedo_fix.txt")
+            ),
+            str(internal.CABLE_AUX_DIR / Path("core/biogeophys/def_soil_params.txt")),
+        ]
```

Comment:
Considering that `internal.CABLE_AUX_DIR` changed to `/g/data/wd9/BenchMarking/CABLE-AUX_v20240122`, we need to  check whether the 2 files are of the correct version.

@ccarouge could you look into this?

Reply:
Actually, we can remove this from `spatial.py` and the tests. It's only needed for CABLE v2. We are now for CABLE v3. If we need a spatial test for v2 then I think it is better to create a payu configuration for it instead of messing with the `crujra_accessN96_1h` config inside `benchcab`.

Reply:
Actually, we can remove this from spatial.py and the tests. It's only needed for CABLE v2. We are now for CABLE v3. If we need a spatial test for v2 then I think it is better to create a payu configuration for it instead of messing with the crujra_accessN96_1h config inside benchcab.